csharp/src/Ice/Internal/RouterInfo.cs

```diff
@@ -146,12 +130,12 @@ public bool addProxy(Ice.ObjectPrx proxy, AddProxyCallback callback)
             }
         }
 
-        _router.addProxiesAsync([proxy]).ContinueWith(
+        _router.addProxiesAsync([new ObjectPrxHelper(reference)]).ContinueWith(
```

Comment:
It would be nice to refactor RouterInfo (reduce callbacks, no ContinueWith), but I didn't want to include too much in this PR.